pkg/vdr/web/resolver.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+)
+
+// Read resolves a did:web did.
+func (v *VDR) Read(didID string, opts ...vdr.ResolveOpts) (*did.Doc, error) {
+	// apply resolve opts
+	docOpts := &vdr.ResolveDIDOpts{}
+
+	for _, opt := range opts {
+		opt(docOpts)
+	}
+
+	defaultClient := &http.Client{}
+	// parse doc url from did
+	url, err := parseDIDWeb(didID)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:web did --> could not parse did:web did --> %w", err)
+	}
+
+	// send get request to retrieve did doc
+	// nolint: errcheck
+	req, _ := http.NewRequest("GET", url, nil)
+
+	var resp *http.Response
+
+	switch docOpts.HTTPClient {
+	case nil:
+		resp, err = defaultClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving did:web did --> error with default get request to %s --> %w", url, err)
+		}
+		// nolint: errcheck
+		defer resp.Body.Close()
+	default:
+		resp, err = docOpts.HTTPClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving did:web did --> error with custom get request to %s --> %w", url, err)
+		}
+		// nolint: errcheck
+		defer resp.Body.Close()
```

Comment:
You can call this line outside of switch statement (it is common in both cases) so there will be only one //nolint: errcheck. We usually don't disable error checks  and you have three of them in this function.